### PR TITLE
Add current_user

### PR DESCRIPTION
**Pull request by vic778**
## Create custom methods

My models included some additional methods.

    Users
        A method that returns the 3 most recent posts for a given user.
    Posts
        A method that updates the posts counter for a user.
        A method which returns the 5 most recent comments for a given post.
    Comments
        A method that updates the comments counter for a post.
    Likes
        A method that updates the likes counter for a post.
    Go to rails c and check if your methods are working.

Additionally, you will need to make current user data available to all controllers.

    Create a method called current_user in your ApplicationController.
    A current_user method should return the first user from a database.
    Don't worry about proper authentication at this time. You will add it in the upcoming projects.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 class Post < ApplicationRecord
-  belongs_to :users
+  belongs_to :user
```
Comment (dasileker) [inline app/models/post.rb]: - [ ] The code should start by declaring a Post model with the following attributes: - name - author_id - created_at - updated_at Next, it declares that this post belongs to an Author, **hint (belong_to => author and class_name to the user);**

- [ ] It also should have many comments and likes. **hint ( has_many => comments and likes)**
- [ ] After saving, the code should increment the posts counter for the author. The update_post_counter method is private and increments a user's posts count after saving their post. The code attempts to increment the author's posts_counter every time they save a post. **hint( post.increment)**

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  validates :text, presence: true, length: { minimum: 5, maximum: 1000 }
```
Comment (dasileker) [inline app/models/comment.rb]: - [ ]  It should declare that this comment belongs to an Author, **hint (belong_to => author and class_name to the user);**
- [ ] It should update_comments_counter method increment the number of comments for the given post.
The code will increment the counter for all posts that belong to the author with a username of "Joe".
The code will increment the counter for all posts that belong to the author with a username of "Joe" and also increment it by one.**hint (post.increment => comments_counter)**

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @vic778,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are passing ✅.
- The PR is good ✅.

## Required Changes ♻️
_Check the comments under the review._

- [ ] Nice job mate however, you haven't added association to the `model/Like.rb` file. it seems you have deleted that file.
- [ ] Please also make sure your ApplicationController contains `#current_user` method. **hint ( to return the first user in the database.)**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by peterrobert — APPROVED** [review]
Hi @vic778, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**